### PR TITLE
Querier : Filters can be saved and loaded to/from XML files

### PR DESCRIPTION
**Pull request by jdenisgiguere**
We can save or load filters to/from XML files. Buttons are added to the filter builder toolbar to make this available.
See https://github.com/georchestra/styler/pull/12 for change on the Styler library side.

This pull request contains a new Doc Service for OGC Filter Encoding. The `allowSaveFilter` property is set in the querier object. The reference to the styler submodule is updated.

### CODE REVIEW

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public FEDocService(final String tempDir, ConnectionPool pgpool) {
+        super(FILE_EXTENSION, MIME_TYPE, tempDir, pgpool);
+    }
+
+    /**
+     * Called before saving the content
+     * @throws DocServiceException
+     */
+    @Override
+    protected void preSave() throws DocServiceException {
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Codacy Issue found: [Document empty method body](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316081044/issues/source?bid=3200181&fileBranchId=3202742#l46)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    }
+
+    /**
+     * Called right after the loading of the file content
+     * @throws DocServiceException
+     */
+    @Override
+    protected void postLoad() throws DocServiceException {
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Codacy Issue found: [Document empty method body](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316081044/issues/source?bid=3200181&fileBranchId=3202742#l55)

File: mapfishapp/src/main/webapp/app/js/GEOR.js [hunk 3]
```diff
@@ -400,7 +400,29 @@ Ext.namespace("GEOR");
             var querierTitle;
             GEOR.querier.events.on({
                 "ready": function(panelCfg) {
+
+                    win = new Ext.Window({
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/js/GEOR.js]: Codacy Issue found: [Variable 'win' implicitly declared.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316076599/issues/source?bid=3200181&fileBranchId=3202742#l404)

File: mapfishapp/src/main/webapp/app/js/GEOR_querier.js [hunk 4]
```diff
@@ -253,6 +253,9 @@ GEOR.querier = (function() {
                     tpl: GEOR.util.getAttributesComboTpl()
                 }
             },
+            toolbarType: 'tbar',
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/js/GEOR_querier.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316076628/issues/source?bid=3200181&fileBranchId=3202742#l256)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java [hunk 5]
```diff
@@ -157,6 +158,11 @@ public void setCredentials(Map<String, UsernamePasswordCredentials> credentials)
      * Absolute (from domain name) URL path where the gml service can be called
      */
     public static final String GML_URL = DOC_URL + "gml/";
+
+    /**
+     * Absolute (from domain name) URL path where the fe service can be called
+     */
+    public static final String FE_URL = DOC_URL + "fe/";
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316081042/issues/source?bid=3200181&fileBranchId=3202742#l165)

File: mapfishapp/src/main/webapp/app/js/GEOR_querier.js [hunk 6]
```diff
@@ -253,6 +253,9 @@ GEOR.querier = (function() {
...
+            allowSaveFilter: true,
+            toolbarType: "tbar",
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/js/GEOR_querier.js]: Codacy Issue found: [Duplicate key 'toolbarType'.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316354646/issues/source?bid=3200181&fileBranchId=3202742#l258)

File: mapfishapp/src/main/webapp/app/js/GEOR_querier.js (hunk 4 above)
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/js/GEOR_querier.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316354646/issues/source?bid=3200181&fileBranchId=3202742#l256)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java (hunk 5 above)
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316359060/issues/source?bid=3200181&fileBranchId=3202742#l165)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java (hunk 2 above)
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Codacy Issue found: [Document empty method body](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316359062/issues/source?bid=3200181&fileBranchId=3202742#l55)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java (hunk 1 above)
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Codacy Issue found: [Document empty method body](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316359062/issues/source?bid=3200181&fileBranchId=3202742#l46)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java (hunk 5 above)
Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Should the code https://github.com/jdenisgiguere/georchestra/blob/FilterBuilder_save_load/mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java#L127-L166 be moved before the first public method declaration?

File: mapfishapp/src/main/webapp/app/js/GEOR_querier.js (hunk 6 above)
Comment (jdenisgiguere) [inline mapfishapp/src/main/webapp/app/js/GEOR_querier.js]: Fixed in 37160b6b9ea4e0c63f31e6618238da0730359d61.

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java (hunk 2 above)
Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Fixed in 8511291f81960dabe1712272b1d48604e2c67a00.

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java (hunk 1 above)
Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/FEDocService.java]: Fixed in 8511291f81960dabe1712272b1d48604e2c67a00.

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java (hunk 5 above)
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/francois-vanderbiest/georchestra/file/2316502841/issues/source?bid=3200181&fileBranchId=3202742#l165)

Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: > Should the code https://github.com/jdenisgiguere/georchestra/blob/FilterBuilder_save_load/mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java#L127-L166 be moved before the first public method declaration?

Yes, I think so.

Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Fixed in 4716a9efad073b2d7021b89a934bde5bd60ec7f4.

Comment (jdenisgiguere) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/DocController.java]: Fixed in 4716a9efad073b2d7021b89a934bde5bd60ec7f4.

**Pull request closed without merging**